cmd/influx/setup.go

```diff
@@ -76,21 +97,47 @@ func getOnboardingRequest() (req *platform.OnboardingRequest, err error) {
 	}
 	req = new(platform.OnboardingRequest)
 	fmt.Println(promptWithColor(`Welcome to InfluxDB 2.0!`, colorYellow))
-	req.User = getInput(ui, "Please type your primary username", "")
-	req.Password = getPassword(ui)
-	req.Org = getInput(ui, "Please type your primary organization name", "")
-	req.Bucket = getInput(ui, "Please type your primary bucket name", "")
-	for {
-		rpStr := getInput(ui, "Please type your retention period in hours (exp 168 for 1 week).\r\nOr press ENTER for infinite.", strconv.Itoa(platform.InfiniteRetention))
-		rp, err := strconv.Atoi(rpStr)
-		if rp >= 0 && err == nil {
-			req.RetentionPeriod = uint(rp)
-			break
+	if setupFlags.username != "" {
+		req.User = setupFlags.username
+	} else {
+		req.User = getInput(ui, "Please type your primary username", "")
+	}
+	if setupFlags.password != "" {
+		req.Password = setupFlags.password
+	} else {
+		req.Password = getPassword(ui)
+	}
+	if setupFlags.token != "" {
+		req.Token = setupFlags.token
+		// else auto-generated by service
+	}
+	if setupFlags.org != "" {
+		req.Org = setupFlags.org
+	} else {
+		req.Org = getInput(ui, "Please type your primary organization name", "")
+	}
+	if setupFlags.bucket != "" {
+		req.Bucket = setupFlags.bucket
+	} else {
+		req.Bucket = getInput(ui, "Please type your primary bucket name", "")
+	}
+	if setupFlags.retention >= 0 {
+		req.RetentionPeriod = uint(setupFlags.retention)
+	} else {
+		for {
+			rpStr := getInput(ui, "Please type your retention period in hours (exp 168 for 1 week).\r\nOr press ENTER for infinite.", strconv.Itoa(platform.InfiniteRetention))
```

Comment:
I know you didn't introduce this line, but `s/exp/ex/`? Or maybe something else that is more clear, or maybe just drop the parenthetical altogether.